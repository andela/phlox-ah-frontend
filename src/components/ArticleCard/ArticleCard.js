import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { Link } from 'react-router-dom';
import {
  Card, CardTitle
} from 'react-materialize';
import './ArticleCard.scss';

let articlePic = '';

export const ArticleCard = (props) => {
  if (props.pic === 'null') {
    articlePic = `https://via.placeholder.com/50?text=AuthorsHaven-${props.title.substring(0, 20)}`;
  } else {
    articlePic = props.pic;
  }
  return (
    <div>
      <Card className={`${props.size} hoverable articleCard`}
        header={
          <CardTitle image={articlePic}>
            <Link key={props.slug} className="capitalize" to={`/articles/${props.slug}`}>
              {props.title.substring(0, 50)}
            </Link>
          </CardTitle>}
        actions={[
          <Link to={'#'} key={props.slug}>Share</Link>,
          <Link key={props.slug} to={`/articles/${props.slug}`}>View Article</Link>,
          <span key={props.slug}>{moment(props.createdAt).format('D MMMM YYYY')}</span>,
          <span key={props.slug}>{props.author}</span>
        ]}>
        <span className="hide-on-large-only">{props.description.substring(0, 120)}</span>
        <span className="hide-on-med-and-down">{props.description.substring(0, 90)}</span>
      </Card>
    </div>);
};

ArticleCard.propTypes = {
  author: PropTypes.string,
  description: PropTypes.string,
  createdAt: PropTypes.string,
  title: PropTypes.string,
  pic: PropTypes.string,
  size: PropTypes.string,
  slug: PropTypes.string
};

export default ArticleCard;
