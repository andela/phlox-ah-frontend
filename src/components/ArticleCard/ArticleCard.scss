@import '../../styles/variables';
.articleCard {
  .tag-div {
    padding-top: 10px;
    display: flex;
    span {
      font-size: 0.90em;
      a {
        color: #00bcd4;
      }
      padding-right: 5px;
      padding-left: 2px;
    }
  }
  a {
    cursor: pointer;
  }
  &.medium {
    height: 350px;
  }
  .card-image {
    max-height: 55%;
    height: 550%;
    box-shadow: 0em .02em rgba(0,0,0,.2);
    img {
      filter: brightness(80%);
    }
    @media #{$mobile} {
      max-height: 48%;
      height: 48%;
    }
    .card-title {
      font: {
        weight: 700;
        size: 1.3em;
      }
      text-align: center;
      a {
        color: #ffffff;
        text-decoration: none;
        &:hover {
          color: #eeecec;
        }
      }
    }
  }
  .card-content {
    div {
      font-size: 0.8em;
      color: #6F6A6A;
      line-height: 22px;
      text-align: justify;
      margin-top: -13px;
    }
  }
  .card-action {
    a {
      font-size: 0.75em;
      color: #3B8CA1;
      margin-right: 14px;
      text-decoration: none;
      &:hover {
        color: #3B8CA1;
      }
    }
    a:not(.btn):not(.btn-large):not(.btn-small):not(.btn-large):not(.btn-floating) {
      color: #3B8CA1;
      &:hover {
        color: rgb(49, 116, 133);
      }
    }
    span {
      font-size: 0.70em;
      color: #ADA9A9;
      padding-right: 0.3em;
    }
  }
}
