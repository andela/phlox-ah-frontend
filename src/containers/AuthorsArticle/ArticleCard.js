import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const ArticleCard = props => (
    <li className="article">
      <div className="photo">
        <img
          src={props.article.imgUrl || props.placeholderImg }
          alt="article-photo"
        />
        <div className="title">
          {props.article.title}
        </div>
      </div>
      <div className="preview">
        {props.article.description}
      </div>
      <div className="footer">
        <span className="share">
          <a href="#">SHARE</a>
        </span>
        <span className="view">
          <Link to={`/articles/${props.article.slug}`}>
            VIEW ARTICLE
          </Link>
        </span>
        <span className="edit">
          <Link to={`/articles/${props.article.slug}/${props.article.status}/edit`}>
            <i className="fas fa-edit"></i>
          </Link>
        </span>
        <span className="delete">
          <button onClick={() => props.deleteArticle(props.article.slug)}>
            <i className="fas fa-trash-alt"></i>
          </button>
        </span>
      </div>
    </li>
);

ArticleCard.propTypes = {
  article: PropTypes.array,
  article.title: PropTypes.string,
  article.imgUrl: PropTypes.string,
  placeholderImg: PropTypes.string,
  article.description: PropTypes.string,
  article.slug: PropTypes.string,
  article.status: PropTypes.string,
  deleteArticle: PropTypes.func
};

export default ArticleCard;
