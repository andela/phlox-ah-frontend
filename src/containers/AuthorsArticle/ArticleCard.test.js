
import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import ArticleCard from './ArticleCard';

const mockStore = configureMockStore([thunk]);

const props = {
  placeholderImg: 'https://via.placeholder.com/300?text=AuthorsHaven',
  article: {
    title: 'title',
    slug: 'title',
    status: 'title',
    description: 'title',
    imgUrl: 'https://via.placeholder.com/300?text=AuthorsHaven',
    body: 'body',
    description: 'description'
  },
  deleteArticle: jest.fn()
};

const store = mockStore({
  ...props
});

let component;
let myComponent;

describe('<ArticleCard />', () => {
  beforeEach(() => {
    component = shallow(
      <ArticleCard {...props} />
    );
  });
  it('should render without throwing an error', () => {
    expect(component).toMatchSnapshot();
  });
  it('should have img tag', () => {
    expect(component.find('img').exists()).toBe(true);
  });
  it('should have div.title tag', () => {
    expect(component.find('div.title').exists()).toBe(true);
  });
  it('should have li.article tag', () => {
    expect(component.find('li.article').exists()).toBe(true);
  });
  it('should have div.photo tag', () => {
    expect(component.find('div.photo').exists()).toBe(true);
  });
  it('should have div.footer tag', () => {
    expect(component.find('div.footer').exists()).toBe(true);
  });
  it('should have span.edit tag', () => {
    expect(component.find('span.edit').exists()).toBe(true);
  });
  it('should have span.view tag', () => {
    expect(component.find('span.view').exists()).toBe(true);
  });
  it('should have span.share tag', () => {
    expect(component.find('span.share').exists()).toBe(true);
  });
  it('should have span.delete tag', () => {
    expect(component.find('span.delete').exists()).toBe(true);
  });
  it('should have Link tag', () => {
    expect(component.find('Link').exists()).toBe(true);
  });
  it('should have i.fas tag', () => {
    expect(component.find('i.fas').exists()).toBe(true);
  });
});
