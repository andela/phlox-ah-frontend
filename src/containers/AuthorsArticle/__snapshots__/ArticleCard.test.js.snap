// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> should render without throwing an error 1`] = `
<li
  className="article"
>
  <div
    className="photo"
  >
    <img
      alt="article-photo"
      src="https://via.placeholder.com/300?text=AuthorsHaven"
    />
    <div
      className="title"
    >
      title
    </div>
  </div>
  <div
    className="preview"
  >
    description
  </div>
  <div
    className="footer"
  >
    <span
      className="share"
    >
      <a
        href="#"
      >
        SHARE
      </a>
    </span>
    <span
      className="view"
    >
      <a>
        VIEW ARTICLE
      </a>
    </span>
    <span
      className="edit"
    >
      <Link
        replace={false}
        to="/articles/title/title/edit"
      >
        <i
          className="fas fa-edit"
        />
      </Link>
    </span>
    <span
      className="delete"
    >
      <button
        onClick={[Function]}
      >
        <i
          className="fas fa-trash-alt"
        />
      </button>
    </span>
  </div>
</li>
`;
